go/kbfs/libkbfs/dir_block_map_disk.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+)
+
+// dirBlockMapDisk tracks dir block info while making a revision, by
+// using a disk-based block cache.
+type dirBlockMapDisk struct {
+	dirtyBcache *DirtyBlockCacheDisk
+	kmd         KeyMetadata
+	ptrs        map[BlockPointer]bool
+}
+
+var _ dirBlockMap = (*dirBlockMapDisk)(nil)
+
+func newDirBlockMapDisk(
+	dirtyBcache *DirtyBlockCacheDisk, kmd KeyMetadata) *dirBlockMapDisk {
+	return &dirBlockMapDisk{
+		dirtyBcache: dirtyBcache,
+		kmd:         kmd,
+		ptrs:        make(map[BlockPointer]bool),
+	}
+}
+
+func (dbmd *dirBlockMapDisk) putBlock(
+	ctx context.Context, ptr BlockPointer, block *DirBlock) error {
+	err := dbmd.dirtyBcache.Put(
```

Comment:
Why does the `dirBlockMapDisk` use the `dirtyBcache` instead of having a second `levelDb`?

Reply:
Entirely to save on code duplication.  `DirtyBlockCacheDisk` already has code to turn in-memory blocks into disk blocks (by "readying" them on puts, and re-assembling them from the encrypted components when reading them back).  Seems like there would be no harm in putting everything together in a single leveldb, so I just went with that approach.